packages/ove-app-networks/src/client/common/networks.js

```diff
@@ -13,6 +13,235 @@ $(function () {
+            default:
+                // The specification is large and we don't support all types of
+                // operators/functions
+                const err = 'Unable to evaluate unknown type: ' + filter.type;
+                log.error(err);
+                throw Error(err);
+        }
+    };
+
+    (function () {
+        // This function resets colors and labels to the original state before
+        // running any further filtering. Edges don't have labels by default and
+        // reducing the parsing overhead can improve performance for some graphs.
+        const reset = function (n, hasLabels) {
+            let ln = n.length;
+            while (ln) {
+                ln--;
+                if (n[ln].origColor === undefined) {
+                    n[ln].origColor = n[ln].color ? n[ln].color : null;
+                } else if (n[ln].origColor === null) {
+                    delete n[ln].color;
+                } else {
+                    n[ln].color = n[ln].origColor;
+                }
+                if (hasLabels) {
+                    if (n[ln].origLabel === undefined) {
+                        n[ln].origLabel = n[ln].label ? n[ln].label : null;
+                    } else if (n[ln].origLabel === null) {
+                        delete n[ln].label;
+                    } else {
+                        n[ln].label = n[ln].origLabel;
+                    }
+                }
+            }
+        };
+        const g = window.ove.context.sigma.graph;
+        reset(g.nodes(), true);
+        reset(g.edges(), false);
+    })();
+
+    // The decision for making edge/node operations are dependent on the presence of the
+    // corresponding filter
+    const nodeFilter = getFromMessage(message, 'node', '$filter');
+    const edgeFilter = getFromMessage(message, 'edge', '$filter');
+
+    let f = (new sigma.plugins.filter(window.ove.context.sigma)).undo();
+    switch (message.operation) {
+        case Constants.Operation.SEARCH:
+            if (nodeFilter) {
+                log.debug('Filtering nodes using filter:', nodeFilter);
+                f = f.nodesBy(function (n) {
+                    return evaluate(n, nodeFilter);
+                });
+            }
+            if (edgeFilter) {
+                log.debug('Filtering edges using filter:', edgeFilter);
+                f.edgesBy(function (n) {
+                    return evaluate(n, edgeFilter);
+                });
+            }
+            f.apply();
+            break;
+        case Constants.Operation.COLOR:
+            if (nodeFilter) {
+                const nodeColor = getFromMessage(message, 'node', 'color');
+                log.debug('Applying color:', nodeColor, 'on all nodes matching filter:', nodeFilter);
+                f = f.nodesBy(function (n) {
+                    if (evaluate(n, nodeFilter)) {
+                        n.color = nodeColor;
+                    }
+                    return true;
```

Comment:
Perhaps add a comment here:

```JavaScript
    return true; // select all nodes, so none are hidden
```
and similarly for the other equivalent statements